src/extension.ts

```diff
@@ -11,7 +11,7 @@ export function activate(context: vscode.ExtensionContext) {
 export function deactivate() {}
 
 
-const authorizationScopes = ['repo', 'workflow'];
+const authorizationScopes = vscode.workspace.getConfiguration('github-graphql-nb').get('scopes') as string[];
```

Comment:
If we read the value of this setting just once, a user would have to reload the VS Code window for the new auth scopes to be picked up. A small suggestion here: in the constructor of the `OctokitController` after line 25 in this file, we could use the `vscode.workspace.onDidChangeConfiguration` event to detect a change to this setting's value, and update the authorization scopes we ask for and clear out the existing auth session:

```ts
vscode.workspace.onDidChangeConfiguration((e) => {
    if (e.affectsConfiguration('github-graphql-nb')) {
        authorizationScopes = vscode.workspace.getConfiguration('github-graphql-nb').get('scopes') ?? []; // Update our cached auth scopes
        this.clearAuthenticationSession(); // Clear out the existing auth session if requested scopes changed
    }
});
```

That way, after you change the value of this setting, you can simply click run and the extension will ask you to reauth with the new scopes before executing your query. This is probably slightly more user friendly than requiring the user to do a full reload. What do you think?

Reply:
I wondered about that. I'll take a look at confit change. Thanks. 😄

Reply:
Added onDigChangeConfiguration.